web/src/main/java/com/bazaarvoice/emodb/web/migrator/LocalRangeMigrator.java

```diff
@@ -56,10 +59,20 @@ public RangeScanUploaderResult migrate(final int taskId, ScanOptions options, fi
 
             Iterator<MigrationScanResult> results = Iterators.limit(allResults, getResplitRowCount(options));
 
+            int maxWritesPerSecond = statusDAO.getMaxWritesPerSecond(migrationId);
+            long maxWritesTimestamp = System.currentTimeMillis();
+
             while (System.currentTimeMillis() - startTime < options.getMaxRangeScanTime().getMillis() && results.hasNext()) {
 
-                Iterator<MigrationScanResult> batchIterator = Iterators.limit(results, BATCH_SIZE);
-                _migratorTools.writeRows(placement, results, maxWritesPerSecond);
+                if (maxWritesTimestamp + (READ_RATE_LIMITER_SECONDS * 1000) > System.currentTimeMillis()) {
```

Comment:
This may be simpler if you use a Guava `Supplier<Integer>` to get the max writes per second.  You could then wrap this with a `Suppliers.memoizeWithExpiration()` to cache the value for a fixed amount of time then refresh automatically.  Going back to the previous comment, the `Supplier` could be what's injected to get the maximum writes per second.

There's kind of an example of this here: https://github.com/bazaarvoice/emodb/blob/master/datacenter/src/main/java/com/bazaarvoice/emodb/datacenter/core/DefaultDataCenters.java#L65